presto-main/src/main/java/com/facebook/presto/operator/window/PagesWindowIndex.java

```diff
@@ -80,6 +82,14 @@ public Slice getSlice(int channel, int position)
     }
 
     @Override
+    public Block getBlock(int channel, int position)
+    {
+        BlockBuilder builder = pagesIndex.getType(channel).createBlockBuilder(new BlockBuilderStatus(), 1);
```

Comment:
I'm going to benchmark it on TPC-DS queries and based on that provide a proper solution.
In a meantime, please don't hesitate to continue reviewing because the potential change will be in the very limited scope (simple change of the method which is called to retrieve block).